### PR TITLE
US Census API utilities

### PR DESCRIPTION
**Pull request by rbhoot**
Scripts helpful to check data availability and download data from US Census API.

### CODE REVIEW

File: scripts/us_census/api_utils/census_api_data_downloader.py [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    status_path = os.path.join(output_path, 'download_status.json')
+
+    if os.path.isfile(status_path):
+        log_list = json.load(open(status_path))
+    else:
+        log_list = []
+    url_list = sync_status_list(log_list, url_list)
+    with open(status_path, 'w') as fp:
+        json.dump(url_list, fp, indent=2)
+    print(len(url_list))
```
Comment (ajaits) [inline scripts/us_census/api_utils/census_api_data_downloader.py:178]: please remove such prints used fro debugging.

File: scripts/us_census/api_utils/census_api_config_fetcher.py [hunk 2]
```diff
@@ -0,0 +1,970 @@
...
+                        for cur_geo in geo_list:
+                            if 'geoLevelDisplay' in cur_geo and 'geoLevelId' in cur_geo and cur_geo[
+                                    'geoLevelDisplay'] != cur_geo['geoLevelId']:
+                                if 'geo_multiple_id' not in error_dict:
+                                    error_dict['geo_multiple_id'] = []
+                                error_dict['geo_multiple_id'].append(
+                                    temp_url + ' ' + cur_geo['name'])
+                                print(cur_geo['name'], 'has multiple geoId ')
+                            # check only 3 key values
+                            if len(cur_geo) > 7:
```
Comment (ajaits) [inline scripts/us_census/api_utils/census_api_config_fetcher.py:640]: Please add a comment describing constants like `7`

File: scripts/us_census/api_utils/census_api_data_downloader.py [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+#         json.dump(url_list, fp, indent=2)
+
+#     # check status before consolidate, warn if any URL status contains fail
+#     if failed_urls_ctr > 0:
+#         logging.warn('%d urls have failed, output files might be missing data.', failed_urls_ctr)
+#     consolidate_files(dataset, table_id, year_list, output_path)
+
+#     end = time.time()
+#     print("The time required to download the", table_id, "dataset :", end-start)
+#     logging.info('The time required to download the %s dataset : %f', table_id, end-start)
```
Comment (ajaits) [inline scripts/us_census/api_utils/census_api_data_downloader.py:447]: Can this block be removed?

File: scripts/us_census/api_utils/url_list_compiler.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    #                 ret_list.append(temp_dict)
+    #     else:
+    #         for i, cur_vars in enumerate(variables_chunked):
+    #             variable_list_str = ','.join(cur_vars)
+    #             temp_dict = {}
+    #             temp_dict['url'] = get_url_variables(dataset, year, 'NAME,' + variable_list_str, geo_str)
+    #             temp_dict['store_path'] = get_file_name_variables(output_path, table_id, year, i, geo_str)
+    #             temp_dict['status'] = 'pending'
+    #             ret_list.append(temp_dict)
+    # return ret_list
```
Comment (ajaits) [inline scripts/us_census/api_utils/url_list_compiler.py:253]: Remove this block if it is not needed?

Comment (rbhoot) [reply scripts/us_census/api_utils/url_list_compiler.py:253]: Have kept the block as it acts as a basic code which needs minor modification. The feature is not useful currently but would be helpful in future and would need minor modifications only. It was hard to implement and test it within the current timeframe.

File: scripts/us_census/api_utils/census_api_data_downloader.py (hunk 3 above)
Comment (rbhoot) [reply scripts/us_census/api_utils/census_api_data_downloader.py:447]: Have kept the block as it acts as a basic code which needs minor modification. The feature is not useful currently but would be helpful in future and would need minor modifications only. It was hard to implement and test it within the current timeframe.

File: scripts/us_census/api_utils/census_api_config_fetcher.py (hunk 2 above)
Comment (rbhoot) [reply scripts/us_census/api_utils/census_api_config_fetcher.py:640]: fixed.

File: scripts/us_census/api_utils/census_api_data_downloader.py (hunk 1 above)
Comment (rbhoot) [reply scripts/us_census/api_utils/census_api_data_downloader.py:178]: fixed.

**Review by ajaits — APPROVED** [review]

**Pull request merged**